cmake/xgboost-config.cmake.in

```diff
@@ -1,5 +1,24 @@
 @PACKAGE_INIT@
 
+include(CMakeFindDependencyMacro)
```

Comment:
Why do we need these?

Reply:
More specifically, I don't think XGBoost binary exposes these dependencies

Reply:
This is a strict rule by CMake: whenever an export target links with a library that's found by `find_package(...)`, the config file *must* specify that library with `find_dependency()`. Otherwise, you'd get an error message like
```
 Target "xgboost" links to target "Threads::Threads" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
```
(from https://github.com/HowardHinnant/date/issues/514)

It does not matter whether XGBoost exposes the dependencies or not.

Reply:
I'm merging this for now, since I'd like to also test #5854. Let me know if you think of a better way of setting up CMake install.